services/QuillLMS/engines/evidence/lib/evidence/evidence/automl_check.rb

```diff
@@ -14,37 +16,73 @@ def initialize(entry, prompt, previous_feedback=[])
 
     # rubocop:disable Metrics/CyclomaticComplexity
     def feedback_object
-      return unless matched_rule
+      return unless matched_automl_rule
 
-      feedback = matched_rule.determine_feedback_from_history(@previous_feedback)
-      highlight = feedback.highlights.map do |h|
+      if matched_low_confidence_rule
```

Comment:
There's a lot of repetition between the `if` and `else` block here that I think is unnecessary.  The only differences are `feedback_type` which I think we can pull from `matched_rule.rule_type`, and the values in `api`.  Speaking of, I think it's pretty weird to save `@confidence_score` as two different values.  We should probably aim to be consistent and just store it as `confidence` in any `FeedbackHistory` that has access to that number so that we can do comparisons between them.  (See my notes below.)

Reply:
@anathomical that makes sense, I'll get to consolidating this on Monday. To your second point, I actually originally had just `confidence` for both, but I think that's a little misleading because it means different things in these cases, since the `@confidence_score` is always from the original model. With just `confidence` for both I'd be worried that someone would read the value as an evaluation of how sure we are about applying the `low_confidence` rule, which it isn't really. I'm open to changing it back but wanted to flag that there is a little bit of disjunction in how we would be understanding the same value in both cases. Let me know what you think.

Reply:
@emilia-friedberg That's a totally valid concern to have.  I think that we probably want comparable values so that, at the very least, someone could do a query of "all `FeedbackHistory` records with confidence = X" and get both items we flagged low confidence and those we didn't.  Especially since, and I just realized this, the spec calls for only using "low confidence" for non-optimal responses.  So we may have a non-trivial number of `AutoML`-type `FeedbackHistory` records that have confidence below the specified threshold.  (I don't think the current code covers this, I'll modify my comment below with a sample I think handles the case pretty simply.)

Basically, you're definitely right that this may cause confusion, but I think we want to be able to compare confidence levels no matter what type of feedback a student sees.